frontend/src/components/App/Home/index.tsx

```diff
@@ -105,18 +124,28 @@ function ContextMenu({ cluster }: { cluster: Cluster }) {
 
       <ConfirmDialog
         open={openConfirmDialog}
-        handleClose={() => setOpenConfirmDialog(false)}
+        handleClose={() => {
+          setOpenConfirmDialog(false);
+        }}
         onConfirm={() => {
           setOpenConfirmDialog(false);
-          removeCluster(cluster);
+          if (cluster.meta_data?.source !== 'dynamic_cluster') {
+            removeCluster(cluster, true);
+          } else {
+            removeCluster(cluster);
+          }
         }}
         title={t('translation|Delete Cluster')}
         description={t(
-          'translation|Are you sure you want to remove the cluster "{{ clusterName }}"?',
+          'translation|This action will delete cluster "{{ clusterName }}" from {{ source }}.',
```

Comment:
Let's make this about the kubeconfig. If a plugin added the cluster, or we loaded it from a file directly, we just want to confirm that the user wants to remove the cluster. The removal from kubeconfig (i.e. non-Headlamp kubeconfigs) should be a configurable option for the user as a checkbox, and that's a very particular case.